risc0/zkvm/src/host/receipt.rs

```diff
@@ -206,14 +214,29 @@ impl SegmentReceipts {
         final_receipt.verify_with_context(ctx)?;
         let metadata = final_receipt.get_metadata()?;
         log::debug!("final: {metadata:#?}");
-        if prev_image_id != metadata.pre.digest() {
+        // If there's more than one segment, the last segment could be the fault checker
```

Comment:
This code seems to work as intended only if the fault checker takes exactly one segment. Does it fit within our minimum po2?

Reply:
I was going to take care of this later but I see that it's not much to implement. It currently fits within a single segment but I've changed the verifier logic to account for the possibility of multiple segments.